src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy

```diff
@@ -597,6 +597,66 @@ class PartialUpdateSpec extends HealthCheckSpecification {
         flow && flowHelperV2.deleteFlow(flow.flowId)
     }
 
+    @Tidy
+    @Tags(LOW_PRIORITY)
+    def "Able to update a flow endpoint for a single-swithc flow using partial update"() {
+        given: "An active single-switch flow"
+        def swPair = topologyHelper.singleSwitchPair
+        def flow = flowHelperV2.randomFlow(swPair)
+        flowHelperV2.addFlow(flow)
+
+        when: "Update the flow: port number and vlan id on the src endpoint"
+        def flowInfoFromDb = database.getFlow(flow.flowId)
+        def ingressCookie = flowInfoFromDb.forwardPath.cookie.value
+        def egressCookie = flowInfoFromDb.reversePath.cookie.value
+        def newPortNumber = topology.getAllowedPortsForSwitch(topology.activeSwitches.find {
+            it.dpId == flow.source.switchId
+        }).last()
+        def newVlanId = flow.source.vlanId + 1
+        flowHelperV2.partialUpdate(flow.flowId, new FlowPatchV2().tap {
+            source = new FlowPatchEndpoint().tap {
+                portNumber = newPortNumber
+                vlanId = newVlanId
+            }
+        })
+
+        then: "Flow is really updated"
+        with(northboundV2.getFlow(flow.flowId)) {
+            it.source.portNumber == newPortNumber
+            it.source.vlanId == newVlanId
+        }
+
+        and: "Flow is valid and pingable"
```

Comment:
you should not really be pinging a single-switch flow. it does nothing

Reply:
'ping' is deleted